unstructured_ingest/v2/processes/connectors/github.py

```diff
@@ -0,0 +1,291 @@
+                        date_created=date_created,
+                        date_processed=str(time()),
+                    ),
+                    additional_metadata={
+                        "content-type": element._headers.get("content-type", ""),
+                        "mode": element._rawData.get("mode", ""),
+                        "type": element._rawData.get("type", ""),
+                        "size": element._rawData.get("size", ""),
+                    },
+                )
+
+
+class GitHubDownloaderConfig(DownloaderConfig):
+    pass
+
+
+@dataclass
+class GitHubDownloader(Downloader):
+    connection_config: GitHubConnectionConfig
+    download_config: GitHubDownloaderConfig
+
+    @requires_dependencies(["github"], extras="github")
+    def _fetch_file(self, path):
+        from github.GithubException import UnknownObjectException
+
+        try:
+            content_file = self.connection_config.get_repo().get_contents(path)
+        except UnknownObjectException:
+            logger.error(f"File doesn't exists {self.connection_config.url}/{path}")
+            return None
+
+        return content_file
+
+    @SourceConnectionNetworkError.wrap
+    @requires_dependencies(["requests"], extras="github")
+    def _fetch_content(self, content_file):
+        import requests
+
+        contents = b""
+        if (
+            not content_file.content  # type: ignore
+            and content_file.encoding == "none"  # type: ignore
+            and content_file.size  # type: ignore
+        ):
+            logger.info("File too large for the GitHub API, using direct download link instead.")
+            # NOTE: Maybe add a raise_for_status to catch connection timeout or HTTP Errors?
+            response = requests.get(content_file.download_url)  # type: ignore
+            if response.status_code != 200:
+                logger.info("Direct download link has failed... Skipping this file.")
+                return None
+            else:
+                contents = response.content
+        else:
+            contents = content_file.decoded_content  # type: ignore
+        return contents
+
+    def _fetch_and_write(self, path) -> None:
+        content_file = self._fetch_file(path)
+        contents = self._fetch_content(content_file)
+        if contents is None:
+            raise ValueError(
+                f"Failed to retrieve file from repo "
+                f"{self.connector_config.url}/{self.path}. Check logs",
+            )
+        with open(path, "wb") as f:
+            f.write(contents)
+
+    @SourceConnectionError.wrap
+    def run(self, file_data: FileData, **kwargs: Any) -> download_responses:
+        download_path = self.get_download_path(file_data=file_data)
+        download_path.parent.mkdir(parents=True, exist_ok=True)
+
+        path = file_data.source_identifiers.fullpath
+        self._fetch_and_write(path)
+
+        return DownloadResponse(file_data=file_data, path=Path(path))
```

Comment:
Will want to wrap this in a `generate_download_response` call: 
```suggestion
        return self.generate_download_response(file_data=file_data, download_path=download_path)
```